pyam/core.py

```diff
@@ -285,12 +287,19 @@ def index(self):
     @property
     def model(self):
         """Return the list of (unique) model names"""
-        return get_index_levels(self.meta, 'model')
+        return self._get_meta_index_levels('model')
 
     @property
     def scenario(self):
         """Return the list of (unique) scenario names"""
-        return get_index_levels(self.meta, 'scenario')
+        return self._get_meta_index_levels('scenario')
+
+    def _get_meta_index_levels(self, name):
+        """Return the list of a level from meta"""
+        if name in self.meta.index.names:
+            return get_index_levels(self.meta, name)
+        # in case of non-standard meta.index.names
+        raise AttributeError(f'Index `{name}` does not exist!')
```

Comment:
```suggestion
        raise KeyError(f'Index `{name}` does not exist!')
```
or maybe `IndexError`

Reply:
I really like the idea of using an `IndexError` when the index dimension doesn't exist, but I think `KeyError` is the more accurate error in this case (per https://docs.python.org/3/library/exceptions.html#IndexError)